graph/src/components/store.rs

```diff
@@ -603,6 +603,13 @@ pub trait Store: Send + Sync + 'static {
     /// Looks up an entity using the given store key.
     fn get(&self, key: EntityKey) -> Result<Option<Entity>, QueryExecutionError>;
 
+    /// Look up multiple entities. Returns a map of entities by type.
+    fn get_many(
+        &self,
+        subgraph_id: &SubgraphDeploymentId,
+        ids_for_type: BTreeMap<&str, Vec<&str>>,
+    ) -> Result<BTreeMap<String, Vec<Entity>>, StoreError>;
```

Comment:
One minor nit: if this returned a `Result<Vec<Entity>, StoreError>`, we could save ourselves building a `BTreeMap` that the caller is not really going to use, anyway. But I doubt it will change things all that much.

Reply:
We need the map key as well when iterating over the value returned.